control_plane/src/storage.rs

```diff
@@ -120,6 +120,8 @@ impl PageServerNode {
         // It takes a while for the page server to start up. Wait until it is
         // open for business.
         for retries in 1..15 {
+            // add a short sleep before first try to avoid spurious error on the first check
```

Comment:
> a short sleep before first try

The comment is puzzling: this sleep is called on every iteration. More importantly, does this really solve anything? If 1 sec is too much of a wait, could we just increase the frequency of polling?

Reply:
Hm, yeah, should move it up out of the loop. Increased polling period doesnt help, because this way there can be even more errors printed before successful startup

Reply:
> because this way there can be even more errors printed before successful startup

Maybe we don't need those errors in the first place? We could replace them with a summary (e.g. "8 tries, success")

Reply:
IMO it is convenient to see them because sometimes I receive something like address already in use and can quickly stop the retries

Reply:
Can we modify pageserver to bind the sockets before daemonizing

Reply:
Nice idea, I thought that it would be messy to pass sync socket to asynchronous hyper server, but I found an API for that, so it feels smooth. Rearranged start_pageserver a bit